src/Common/src/Internal/Text/Utf8String.cs

```diff
@@ -112,5 +112,41 @@ public bool Equals(Utf8String other)
                 }
             }
         }
+
+        public int CompareTo(object other)
+        {
+            if (!(other is Utf8String))
+                throw new ArgumentException(nameof(other));
+
+            return Compare(this, (Utf8String)other);
+        }
+
+        private static int Compare(Utf8String strA, Utf8String strB)
+        {
+            int length = Math.Min(strA.Length, strB.Length);
+
+            unsafe
```

Comment:
For the unsafe code, I think it might be interesting to convert to Span<byte>, but I'm not convinced that needs to be done before checkin. unsafe is the pattern used elsewhere in this class, and this isn't changing that concept.

Reply:
I'll leave it as unsafe for this check-in. In the future we should overhaul the class to use Span - there's a comment at the top stating we should do that.